integration-tests/src/test/java/com/datastax/oss/driver/api/core/type/codec/registry/CodecRegistryIT.java

```diff
@@ -170,4 +185,171 @@ public void should_be_able_to_register_and_use_custom_codec() {
       assertThat(row.getFloat(0)).isEqualTo(3.0f);
     }
   }
+
+  // TODO: consider moving this into source as it could be generally useful.
+  private abstract static class MappingCodec<O, I> implements TypeCodec<O> {
```

Comment:
I followed `OptionalCodec`s implementation in java driver extras, which also used `MappingCodec` which is generally useful for going from one java type to another, not sure if we wanted something like this in the source path, I think at this time, probably not, but might be nice if we decide to add additional mapped codecs.